buildSrc/build.gradle.kts

```diff
@@ -275,3 +277,16 @@ fun Project.applyKotlinProjectConventions() {
     }
 }
 
+fun failOnEmptySourceFolder() {
```

Comment:
Can we somehow share this method via a precompiled script plugin between `buildSrc` and the root build?

Reply:
You can find a sample here: https://github.com/gradle/gradle/pull/14472

We probably shouldn't use an included build for this logic, since configuration caching does not support that, yet.

Reply:
Lets do it like this for now, and share the logic once we can use an included build.
I'll add this topic here.

https://github.com/gradle/gradle-private/issues/3139

Reply:
What about sharing the `kts` script in the `gradle/shared-with-buildSrc` directory?

Also, why don't we just fix the cache miss for everyone? :D